docs/tfengine/recipes/deployment.md

```diff
@@ -0,0 +1,113 @@
+# Terraform Deployment Recipe
+
+## Properties
+
+### enable_terragrunt
+
+Whether to convert to a Terragrunt deployment.
+Adds a terragrunt.hcl file in the deployment.
+
+Type: boolean
+
+### state_bucket
+
+State bucket to use for GCS backend. Does nothing if 'enable_terragrunt' is true.
+
+Type: string
+
+### state_path_prefix
+
+Object path prefix for GCS backend. Does nothing if 'enable_terragrunt' is true.
+
+Type: string
+
+### terraform_addons
+
+Extra addons to set in the deployment.
+
+Type: object
+
+### terraform_addons.deps
+
+Additional dependencies on other deployments.
+
+Type: array(object)
+
+### terraform_addons.deps.mock_outputs
+
+Mock outputs for the deployment to add.
+
+Type: object
```

Comment:
From the way it is right now, it's not clear what the object structure is. I wouldn't know how to populate mock_outputs without seeing an example.

Reply:
Not just this one, in general any "object" type.

Reply:
Do you have any suggestions on how to improve it? I was thinking of maybe adding some indentation etc. For most objects, the following fields are the 'inner' fields e.g. 

```
## cicd

Config for CICD. If unset there will be no CICD.

Type: object

### cicd.apply_trigger

Config block for the postsubmit apply/deployyemt Cloud Build trigger.
If specified,create the trigger and grant the Cloud Build Service Account
necessary permissions to perform the build.

...
```

Reply:
Indentation would help, yes. Try it so I can see how it looks.